mainsite/selling/forms.py

```diff
@@ -35,6 +36,51 @@ def __init__(self, *args, **kwargs):
         #Gets all the available categories for the new item
         self.fields['category'].queryset = Category.objects.all()
 
+    def clean(self):
+        super().clean()
+
+
+        uplPos = self._request.POST.getlist('uplPos').copy()
+        currUpl = self._request.POST.getlist('currUpl').copy()
+        cleanedPics = []
+        validationErrors = []
+        for pic in self._request.FILES.getlist('curr_picture'):
+            # please, god, deliver these unto me in source ordering
+            #  despite not mentioning this feature anywhere in your holy texts, the Django docs,
+            #   which I guess in this metaphor makes StackExchange the equivalent of rabbinic commentaries
+            try:
+                result = self._clean_a_picture(pic)
+                if result is not None:
+                    cleanedPics.append(result)
+            except forms.ValidationError as e:
+                validationErrors.append(e)
+        if len(validationErrors) > 0:
+            raise forms.ValidationError(validationErrors)
+        if len(cleanedPics) + len(currUpl) == 0:
+            raise forms.ValidationError( "At least one picture is needed!", code="empty")
+
+        item = self.instance  # ?
+
+        self.otherPictures = item.pictures
+        self.specialPictures = []
+        self.cleaned_data["pictures"] = []
+        i = 1
+        catpic = None
+        while(1):
+            if len(uplPos) > 0 and str(i) == uplPos[0]:
+                uplPos.pop(0)
```

Comment:
Might make more sense to split this up into two loops; one that iterates over `uplPos` (`for pos in uplPos`) and one that adds any additional pictures after that. 

I'm also somewhat confused why it's possible for `len(uplPos) != len(cleanedPics)`.

Reply:
The loops are intertwined -- uplPos is *only* for the preexisting images, which are already in the database. So the loops are combined because the position tracker i is updated after each step -- each loop *either* pulls in an already-uploaded image, *or* generates a new CatalogItemPicture, but either way the relevant position is assigned and i is incremented.
I've added the above comment to the code for future reference, it probably should've been commented the first time.

Reply:
So to be more clear on the different vars involved: cleanedPics should contain all new pics being added to the database; uplPos contains the positions of already-existing images and currUpl contains the pks of those preexisting images.

Reply:
Makes sense.